frontend/packages/rhoas-plugin/src/components/service-table/ServiceInstanceTable.tsx

```diff
@@ -51,6 +58,50 @@ const ServiceInstanceTable: React.FC<ServiceInstanceTableProps> = ({
   const [sortBy, setSortBy] = React.useState({});
   const { t } = useTranslation();
 
+  const returnIconStatus = (status: string) => {
```

Comment:
This is causing render loop as dependencies change.
Move it inside the useCallback callback. Alternatively, wrap the 'returnIconStatus' definition into its own useCallback() Hook

Reply:
returnIconStatus is not best name. 

This doesn't need to be part of the rendering code at all. It can be a separate component called ServiceIconStatus

Reply:
moved to separate file